Expose DecodedImage in the public API

calebfletcher opened the pull request:
The [`RenderBundle::add_image`](https://docs.rs/galileo/latest/galileo/render/render_bundle/struct.RenderBundle.html#method.add_image) method takes in a `DecodedImage`, but this struct is not exposed to the user. The only way to get a `DecodedImage` in the current API is to construct a `UrlImageProvider` and load everything through cached URLs, but if you're just looking to load a local file (similar to how the georust example uses a local icon`) then this is a bit unwieldy.

This PR makes the `DecodedImage` struct public, and adds the documentation required as it is now user-facing.

I would be open to suggestions on other ways to fix this issue, such as making `RenderBundle::add_image` take in an `image::RgbaImage` directly and remove the need for the DecodedImage struct entirely.

Maximkaaa commented on galileo/src/decoded_image.rs:
```diff
@@ -1,15 +1,22 @@
+/// An image that has been loaded into memory.
 #[derive(Debug, Clone)]
 pub struct DecodedImage {
+    /// raw bytes of the image, in RGBA order.
```
```suggestion
    /// Raw bytes of the image, in RGBA order.
```

Maximkaaa commented:
I don't think `RenderBundle` can take charge or decoding an image since the same image can be used in more than one bundles. For example, if a point on a vector tile layer is rendered as an icon, every tile is renderred into separate bundle, but we want to decode the icon only once. So we have to take decoded image here.

One thing that may be improved is it is usually not a good thing to expose fields of a struct in public API, so it would probably be better to make them at least `pub(crate)`, and provide methods to construct the structure without access to the fields.

Maximkaaa commented:
Clippy error is due to Rust update. Please, fix it if you don't mind. Otherwise your PR would have to wait until I'm done with my text rendering rabbit hole...

calebfletcher commented:
> I don't think `RenderBundle` can take charge or decoding an image since the same image can be used in more than one bundles. For example, if a point on a vector tile layer is rendered as an icon, every tile is renderred into separate bundle, but we want to decode the icon only once. So we have to take decoded image here.

Yeah definitely wouldn't want the RenderBundle to do the decoding, but the `image::RgbaImage` is already decoded (and has nearly the exact same struct layout as the `DecodedImage` here. Happy to keep it how it is though.

> One thing that may be improved is it is usually not a good thing to expose fields of a struct in public API, so it would probably be better to make them at least `pub(crate)`, and provide methods to construct the structure without access to the fields.

Good point, fixed, and I've added a constructor that takes in a pixel buffer and dimensions in case the user already has an image in memory or that hasn't come from a file.

> Clippy error is due to Rust update. Please, fix it if you don't mind. Otherwise your PR would have to wait until I'm done with my text rendering rabbit hole...

Fixed that clippy lint, once this PR is merged I'll rebase my other PR so CI passes there too.
